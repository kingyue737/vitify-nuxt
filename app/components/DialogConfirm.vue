<script setup lang="ts">
const dialog = ref(false)
const confirmed = ref(false)
let resolve: (value: boolean) => void
const message = ref('')
watch(dialog, (v) => {
  if (!v) {
    resolve(confirmed.value)
  }
})
function open(text: string) {
  confirmed.value = false
  dialog.value = true
  message.value = text
  return new Promise<boolean>((resolveFn) => {
    resolve = resolveFn
  })
}
function confirm() {
  confirmed.value = true
  dialog.value = false
}
function cancel() {
  confirmed.value = false
  dialog.value = false
}
defineExpose({ open })
</script>

<template>
  <v-dialog v-model="dialog" max-width="400px">
    <v-card>
      <v-card-text class="font-weight-bold d-flex">
        <v-icon class="mr-2" color="warning" icon="$warning" />
        <span>{{ message }}</span>
      </v-card-text>
      <v-card-actions>
        <v-spacer />
        <v-btn color="primary" @click="cancel"> Cancel </v-btn>
        <v-btn color="primary" @click="confirm"> Confirm </v-btn>
        <v-spacer />
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>
