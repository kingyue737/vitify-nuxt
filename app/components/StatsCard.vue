<script setup lang="ts">
withDefaults(
  defineProps<{
    icon: string
    iconClass?: string
    color: string
    title: string
    value: number | null
    unit?: string
    formatter?: (v: number) => string
  }>(),
  {
    iconClass: '',
    value: null,
    unit: '',
    formatter: (v: number) => v.toString(),
  },
)
</script>

<template>
  <v-card class="stats-card v-alert--border-top">
    <v-icon
      size="x-large"
      class="stats-icon"
      :color="color"
      :class="iconClass"
      :icon="icon"
    />
    <div class="card-title ml-auto text-right">
      <span
        class="card-title--name font-weight-bold"
        :class="`text-${color}`"
        v-text="title"
      />
      <h3
        class="font-weight-regular d-inline-block ml-2"
        style="font-size: 18px"
      >
        {{ value != null ? formatter(value) : '' }}
        <small v-if="unit">{{ unit }}</small>
      </h3>
      <v-divider />
    </div>
    <div class="v-alert__border" :class="`text-${color}`" />
    <div
      v-if="$slots.footer"
      class="text-grey text-right stats-footer text-caption"
    >
      <slot name="footer" />
    </div>
  </v-card>
</template>

<style scoped>
.stats-card {
  padding: 5px;
  padding-top: 10px;
  .card-title {
    width: fit-content;
    .card-title--name {
      display: inline-block;
      backdrop-filter: blur(3px);
    }
  }
  .caption {
    font-size: 12px;
    letter-spacing: 0;
  }
  .stats-icon {
    position: absolute;
    opacity: 0.3;
  }
  .stats-footer {
    :deep(span) {
      display: inline-block;
      font-size: 12px !important;
      letter-spacing: 0 !important;
    }
  }
}
</style>
