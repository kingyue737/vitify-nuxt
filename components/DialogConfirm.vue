<script setup lang="ts">
const dialog = ref(false)
const confirmed = ref(false)
let resolve: (value: boolean) => void
const message = ref('')
watch(dialog, (v) => {
  if (!v) {
    resolve(confirmed.value)
  }
})
function open(text: string) {
  confirmed.value = false
  dialog.value = true
  message.value = text
  return new Promise<boolean>((resolveFn) => {
    resolve = resolveFn
  })
}
function confirm() {
  confirmed.value = true
  dialog.value = false
}
function cancel() {
  confirmed.value = false
  dialog.value = false
}
defineExpose({ open })
</script>

<template>
  <v-dialog v-model="dialog" max-width="400px">
    <v-card style="z-index: -1">
      <v-card-text class="font-weight-bold d-flex">
        <v-icon class="mr-2" color="warning">$warning</v-icon>
        <span>{{ message }}</span>
      </v-card-text>
      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn color="primary darken-1" @click="cancel">Cancel</v-btn>
        <v-btn color="primary darken-1" @click="confirm">Confirm</v-btn>
        <v-spacer></v-spacer>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>
