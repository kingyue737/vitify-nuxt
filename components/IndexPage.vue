<script setup lang="ts">
const route = useRoute()

const items = computed(() =>
  route.matched
    .filter((v) => v.path === route.path)[0]
    .children.filter((c) => c.path)
    .toSorted(
      (a, b) => (a.meta?.drawerIndex ?? 99) - (b.meta?.drawerIndex ?? 98),
    )
    .map((c) => ({
      title: c.meta?.title,
      to: c.name ? c : `${route.path}/${c.path}`,
      prependIcon: c.meta?.icon,
      subtitle: c.meta?.subtitle,
    })),
)
</script>

<template>
  <v-container>
    <v-row>
      <v-col>
        <v-card v-for="item in items" :key="item.title" class="mb-1">
          <v-list-item
            v-bind="item"
            append-icon="mdi-chevron-right"
            :ripple="false"
            class="py-4"
          />
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>
