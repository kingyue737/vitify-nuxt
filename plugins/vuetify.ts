import type { IconProps } from 'vuetify'
import { Icon } from '#components'
import type { VDataTable } from 'vuetify/components'
import { useStorage } from '@vueuse/core'
import { aliases } from 'vuetify/iconsets/mdi'

export type DataTableHeaders = VDataTable['$props']['headers']

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.hook('vuetify:configuration', ({ vuetifyOptions }) => {
    vuetifyOptions.icons = {
      defaultSet: 'nuxtIcon',
      sets: {
        nuxtIcon: {
          component: ({ icon, tag, ...rest }: IconProps) =>
            h(tag, rest, [h(Icon, { name: aliases[icon as string] ?? icon })]),
        },
      },
      aliases,
    }
    const primary = useStorage('theme-primary', '#1697f6').value
    vuetifyOptions.theme = {
      themes: {
        light: { colors: { primary } },
        dark: { colors: { primary } },
      },
    }
  })
})
