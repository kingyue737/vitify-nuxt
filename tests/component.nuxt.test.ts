import { describe, expect, it } from 'vitest'
import { mountSuspended } from '@nuxt/test-utils/runtime'
import DialogConfirm from '~/components/DialogConfirm.vue'
import { VCard } from 'vuetify/components'

describe('component DialogConfirm.vue', () => {
  it('should not open', async () => {
    const wrapper = await mountSuspended(DialogConfirm)
    expect(wrapper.isVisible()).toBe(false)
  })

  it('should open and close', async () => {
    const wrapper = await mountSuspended(DialogConfirm)
    wrapper.vm.open('message')
    await nextTick()
    const cardWrapper = wrapper.getComponent(VCard)
    expect(cardWrapper.text()).toContain('message')
    expect(cardWrapper.html()).toMatchSnapshot()
    expect(cardWrapper.find('button').exists()).toBe(true)

    await cardWrapper.find('button').trigger('click')
    expect(cardWrapper.isVisible()).toBe(false)
  })
})
