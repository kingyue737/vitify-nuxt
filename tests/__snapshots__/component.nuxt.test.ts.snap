// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component DialogConfirm.vue > should open and close 1`] = `
"<div class="v-card v-card--flat v-theme--light v-card--border v-card--density-default rounded-lg v-card--variant-elevated" style="z-index: -1;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background bg-primary" style="width: 100%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long bg-primary"></div>
          <div class="v-progress-linear__indeterminate short bg-primary"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class="v-card-text font-weight-bold d-flex"><i class="v-icon notranslate v-theme--light v-icon--size-default text-warning mr-2" aria-hidden="true"><svg class="v-icon__svg" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true">
        <path d="M13,13H11V7H13M13,17H11V15H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"></path>
      </svg></i><span>message</span></div>
  <div class="v-card-actions">
    <div class="v-spacer"></div><button type="button" class="v-btn v-btn--slim v-theme--light text-primary darken-1 v-btn--density-default rounded-xl v-btn--size-default v-btn--variant-text"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Cancel</span>
      <!---->
      <!---->
    </button><button type="button" class="v-btn v-btn--slim v-theme--light text-primary darken-1 v-btn--density-default rounded-xl v-btn--size-default v-btn--variant-text"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Confirm</span>
      <!---->
      <!---->
    </button>
    <div class="v-spacer"></div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
